src/libraries/System.Private.CoreLib/src/System/Globalization/CompareInfo.OSX.cs

```diff
@@ -32,6 +37,67 @@ private unsafe int CompareStringNative(ReadOnlySpan<char> string1, ReadOnlySpan<
             return result;
         }
 
+        private unsafe int IndexOfCoreNative(ReadOnlySpan<char> source, ReadOnlySpan<char> target, CompareOptions options, int* matchLengthPtr, bool fromBeginning)
+        {
+            Debug.Assert(!GlobalizationMode.Invariant);
+            Debug.Assert(!target.IsEmpty);
+            Debug.Assert(!GlobalizationMode.UseNls);
+            Debug.Assert(target.Length != 0);
+
+            if (_isAsciiEqualityOrdinal && CanUseAsciiOrdinalForOptions(options))
```

Comment:
Will `options` ever have the `CompareOptions.IgnoreSymbols` bit set on OSX? `IgnoreSymbols` isn't supported on OSX, will that be caught before this function is hit?

Reply:
But if we end up using the ICU code path, i dont think it would add enough value to warrant removing `CanUseAsciiOrdinalForOptions` under a Hybrid guard.

Reply:
Refactored this part.